src/status_im/browser/core.cljs

```diff
@@ -274,19 +275,58 @@
               (navigation/navigate-to-cofx :browser nil)
               (resolve-url nil))))
 
+(fx/defn web3-error-callback
+  {:events [:browser.dapp/transaction-on-error]}
+  [{{:keys [webview-bridge]} :db} message-id message]
+  {:browser/send-to-bridge
+   {:message {:type      constants/web3-send-async-callback
+              :messageId message-id
+              :error     message}
+    :webview webview-bridge}})
+
+(fx/defn dapp-complete-transaction
+  {:events [:browser.dapp/transaction-on-result]}
+  [{{:keys [webview-bridge]} :db} message-id id result]
+  ;;TODO check and test id
+  {:browser/send-to-bridge
+   {:message {:type      constants/web3-send-async-callback
+              :messageId message-id
+              :result    {:jsonrpc "2.0"
+                          :id      (int id)
+                          :result  result}}
+    :webview webview-bridge}})
+
+(defn normalize-sign-message-params
+  "NOTE (andrey) we need this function, because params may be mixed up,
+  so we need to figure out which one is address and which message"
+  [params]
+  (let [first_param           (first params)
```

Comment:
sorry for neatpicking, but why `_` in names, but not `-`?

Reply:
this is something old, just moved here, but agree it's weird